src/pybind/mgr/prometheus/module.py

```diff
@@ -321,10 +321,22 @@ def get_metadata_and_osd_status(self):
                     " and metadata records for this osd".format(id_)
                 )
                 continue
-            dev_class = next((osd for osd in osd_devices if osd['id'] == id_))
+
+            dev_class = None
+            for osd_device in osd_devices:
+                if osd['id'] == id_:
```

Comment:
should be `osd_device`, i guess?

```
2018-03-01T04:23:16.497 INFO:tasks.mgr.test_prometheus:/: 200 (155 bytes)
2018-03-01T04:23:16.504 INFO:tasks.ceph.mgr.x.smithi162.stdout:::ffff:172.21.0.51 - - [01/Mar/2018:04:23:16] "GET / HTTP/1.1" 200 155 "" "python-requests/2.12.5"
2018-03-01T04:23:16.507 INFO:tasks.ceph.mgr.x.smithi162.stderr:[01/Mar/2018:04:23:16] HTTP Traceback (most recent call last):
2018-03-01T04:23:16.508 INFO:tasks.ceph.mgr.x.smithi162.stderr:  File "/usr/lib/python2.7/dist-packages/cherrypy/_cprequest.py", line 670, in respond
2018-03-01T04:23:16.508 INFO:tasks.ceph.mgr.x.smithi162.stderr:    response.body = self.handler()
2018-03-01T04:23:16.508 INFO:tasks.ceph.mgr.x.smithi162.stderr:  File "/usr/lib/python2.7/dist-packages/cherrypy/lib/encoding.py", line 217, in __call__
2018-03-01T04:23:16.508 INFO:tasks.ceph.mgr.x.smithi162.stderr:    self.body = self.oldhandler(*args, **kwargs)
2018-03-01T04:23:16.508 INFO:tasks.ceph.mgr.x.smithi162.stderr:  File "/usr/lib/python2.7/dist-packages/cherrypy/_cpdispatch.py", line 61, in __call__
2018-03-01T04:23:16.508 INFO:tasks.ceph.mgr.x.smithi162.stderr:    return self.callable(*self.args, **self.kwargs)
2018-03-01T04:23:16.508 INFO:tasks.ceph.mgr.x.smithi162.stderr:  File "/usr/lib/ceph/mgr/prometheus/module.py", line 446, in metrics
2018-03-01T04:23:16.508 INFO:tasks.ceph.mgr.x.smithi162.stderr:    metrics = global_instance().collect()
2018-03-01T04:23:16.509 INFO:tasks.ceph.mgr.x.smithi162.stderr:  File "/usr/lib/ceph/mgr/prometheus/module.py", line 382, in collect
2018-03-01T04:23:16.509 INFO:tasks.ceph.mgr.x.smithi162.stderr:    self.get_metadata_and_osd_status()
2018-03-01T04:23:16.509 INFO:tasks.ceph.mgr.x.smithi162.stderr:  File "/usr/lib/ceph/mgr/prometheus/module.py", line 327, in get_metadata_and_osd_status
2018-03-01T04:23:16.509 INFO:tasks.ceph.mgr.x.smithi162.stderr:    if osd['id'] == id_:
2018-03-01T04:23:16.509 INFO:tasks.ceph.mgr.x.smithi162.stderr:KeyError: 'id'
```

http://pulpito.ceph.com/kchai-2018-03-01_03:47:54-rados-wip-kefu-testing-2018-03-01-1006-distro-basic-smithi/2237480/

Reply:
oops, you're right, updated.